### PR TITLE
MQTT plugin

### PR DESCRIPTION
**Pull request by Perforex**
Changes made. Many thanks to James for the testing and suggestions.

### CODE REVIEW

File: mqttplugin.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+""" Fauxmo plugin for simple MQTT requests. Requires paho-mqtt v1.3.1 (https://www.eclipse.org/paho/clients/python/docs/)
```
Comment (n8henrie) [inline mqttplugin.py:1]: Please follow [the example plugins](https://github.com/n8henrie/fauxmo-plugins/blob/aa98f74c2eb3204fb0270284b95c269aa950670e/hassapiplugin.py#L40) regarding dependencies so we can keep things consistent, as recommended [in the readme](https://github.com/n8henrie/fauxmo-plugins#plugins-provided-in-the-fauxmo-plugins-repo).

**Review by n8henrie — CHANGES_REQUESTED** [review]
Lots of linter errors, will need to make sure `tox` passes before merging.

```
./mqttplugin.py:1:1: D210 No whitespaces allowed surrounding docstring text
./mqttplugin.py:1:1: D400 First line should end with a period
./mqttplugin.py:1:80: E501 line too long (121 > 79 characters)
./mqttplugin.py:3:80: E501 line too long (85 > 79 characters)
./mqttplugin.py:3:86: W291 trailing whitespace
./mqttplugin.py:4:80: E501 line too long (91 > 79 characters)
./mqttplugin.py:4:92: W291 trailing whitespace
./mqttplugin.py:5:80: E501 line too long (91 > 79 characters)
./mqttplugin.py:5:92: W291 trailing whitespace
./mqttplugin.py:6:80: E501 line too long (90 > 79 characters)
./mqttplugin.py:9:80: E501 line too long (88 > 79 characters)
./mqttplugin.py:10:80: E501 line too long (87 > 79 characters)
./mqttplugin.py:36:80: E501 line too long (96 > 79 characters)
./mqttplugin.py:36:97: W291 trailing whitespace
./mqttplugin.py:37:80: E501 line too long (93 > 79 characters)
./mqttplugin.py:38:52: W291 trailing whitespace
./mqttplugin.py:44:1: I100 Import statements are in the wrong order. import logging should be before import paho.mqtt.client
./mqttplugin.py:44:1: I201 Missing newline before sections or imports.
./mqttplugin.py:45:1: I201 Missing newline before sections or imports.
./mqttplugin.py:46:1: F401 'fauxmo.__version__' imported but unused
./mqttplugin.py:46:1: F401 'fauxmo.logger' imported but unused
./mqttplugin.py:46:1: I100 Import statements are in the wrong order. from fauxmo should be before from fauxmo.plugins
./mqttplugin.py:48:1: E302 expected 2 blank lines, found 1
./mqttplugin.py:48:1: D101 Missing docstring in public class
./mqttplugin.py:50:1: D107 Missing docstring in __init__
./mqttplugin.py:57:17: E231 missing whitespace after ':'
./mqttplugin.py:68:35: E231 missing whitespace after ','
./mqttplugin.py:70:1: D102 Missing docstring in public method
./mqttplugin.py:70:24: E231 missing whitespace after ','
./mqttplugin.py:71:20: E225 missing whitespace around operator
./mqttplugin.py:73:23: E225 missing whitespace around operator
./mqttplugin.py:73:26: E701 multiple statements on one line (colon)
./mqttplugin.py:73:39: E225 missing whitespace around operator
./mqttplugin.py:74:23: E225 missing whitespace around operator
./mqttplugin.py:74:26: E701 multiple statements on one line (colon)
./mqttplugin.py:74:39: E225 missing whitespace around operator
./mqttplugin.py:79:1: D102 Missing docstring in public method
./mqttplugin.py:81:33: E701 multiple statements on one line (colon)
./mqttplugin.py:84:36: E231 missing whitespace after ','
./mqttplugin.py:84:47: E231 missing whitespace after ','
./mqttplugin.py:85:38: E231 missing whitespace after ','
./mqttplugin.py:85:54: E703 statement ends with a semicolon
./mqttplugin.py:86:28: E703 statement ends with a semicolon
./mqttplugin.py:90:1: D102 Missing docstring in public method
./mqttplugin.py:92:33: E701 multiple statements on one line (colon)
./mqttplugin.py:95:36: E231 missing whitespace after ','
./mqttplugin.py:95:47: E231 missing whitespace after ','
./mqttplugin.py:96:39: E231 missing whitespace after ','
./mqttplugin.py:96:56: E703 statement ends with a semicolon
./mqttplugin.py:97:28: E703 statement ends with a semicolon
./mqttplugin.py:101:1: D102 Missing docstring in public method
./mqttplugin.py:103:35: E701 multiple statements on one line (colon)
./mqttplugin.py:104:33: E701 multiple statements on one line (colon)
./mqttplugin.py:108:36: E231 missing whitespace after ','
./mqttplugin.py:110:28: W291 trailing whitespace
./mqttplugin.py:114:1: D102 Missing docstring in public method
./mqttplugin.py:116:9: F811 redefinition of unused 'logger' from line 46
./mqttplugin.py:117:34: W291 trailing whitespace
./mqttplugin.py:119:25: W291 trailing whitespace
./mqttplugin.py:121:32: W291 trailing whitespace
./mqttplugin.py:123:25: W291 trailing whitespace
./mqttplugin.py:125:20: W291 trailing whitespace
```

**Comment by n8henrie** [comment]
I added the linter tests to Travis (instead of just running locally), so you'll need to `pull` that in. New file (`.travis.yml`) so shouldn't have any merge conflicts.

**Review by n8henrie — APPROVED** [review]

**Review by n8henrie — CHANGES_REQUESTED** [review]
Please see review above re: linter changes

**Review by greypanda — CHANGES_REQUESTED** [review]
This is working well. I wish there was a way to keep the mqtt loop running so you don't have to do a connect for every command.

**Comment by n8henrie** [comment]
Merged into dev with https://github.com/n8henrie/fauxmo-plugins/commit/d2cfea32e38359b77d58e2db75062a1a18617335

**Pull request closed without merging**